### PR TITLE
Update Tracy to 5a36782c (2022-08-01)

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

**Review by benvanik — APPROVED** [review]

**Review by ScottTodd — APPROVED** [review]

**Comment by antiagainst** [comment]
@benvanik: There are some plotting changes incorporated into this one, see the release notes [here](https://github.com/wolfpld/tracy/compare/80f6a93da79af94fc4604e37439f5a75d5e7dfbd...5a36782c83ce9fb8ab3ae0f8f018da21e8681eff#diff-7ee66c4f1536ac84dc5bbff1b8312e2eef24b974b3e48a5c5c2bcfdf2eb8f3ce and the manual [here](https://github.com/wolfpld/tracy/commit/347317745b725c267ff21b1a16e325b46d242ac3). I just did mechanical change to make it compile; you might want to see if they are interesting features to be used! :)

**Comment by benvanik** [comment]
oh wow, those are some fantastic changes to plotting! I'll definitely play with those!

File: build_tools/third_party/tracy/CMakeLists.txt [hunk 1]
```diff
@@ -115,7 +115,7 @@ endfunction()
 # Common library
 #-------------------------------------------------------------------------------
 
-file(GLOB COMMON_SRCS ${TRACY_SOURCE_DIR}/common/*.cpp)
+file(GLOB COMMON_SRCS ${TRACY_SOURCE_DIR}/public/common/*.cpp)
```
Comment (ScottTodd) [inline build_tools/third_party/tracy/CMakeLists.txt:118]: Not sure if this is a flake, but something in Tracy errored on our release build: https://github.com/iree-org/iree/runs/7629125198?check_suite_focus=true

```
  [274/348] Linking CXX executable tools/iree-run-module
  [275/348] Linking CXX executable tools/iree-run-trace
  [276/348] Building CXX object tracy/CMakeFiles/IREETracyCommon.dir/__/__/__/third_party/tracy/public/common/TracyStackFrames.cpp.o
  [277/348] Building CXX object tracy/CMakeFiles/IREETracyCommon.dir/__/__/__/third_party/tracy/public/common/TracySystem.cpp.o
  [278/348] Building CXX object tracy/CMakeFiles/IREETracyCommon.dir/__/__/__/third_party/tracy/public/common/TracySocket.cpp.o
  [279/348] Building CXX object runtime/bindings/python/CMakeFiles/iree_runtime_bindings_python_PyExtRt.dir/vm.cc.o
  In file included from /main_checkout/iree/runtime/bindings/python/vm.cc:7:
  /main_checkout/iree/runtime/bindings/python/./vm.h: In static member function ‘static void iree::python::VmRef::BindRefProtocol(PyClass&, TypeIdFunctor, RetainRefFunctor, DerefFunctor, IsaFunctor)’:
  /main_checkout/iree/runtime/bindings/python/./vm.h:213:11: warning: typedef ‘using RawPtrType = typename PyClass::type::RawPtrType’ locally defined but not used [-Wunused-local-typedefs]
    213 |     using RawPtrType = typename WrapperType::RawPtrType;
        |           ^~~~~~~~~~
  [280/348] Building CXX object tracy/CMakeFiles/IREETracyCommon.dir/__/__/__/third_party/tracy/public/common/tracy_lz4hc.cpp.o
  [281/348] Building CXX object tracy/CMakeFiles/IREETracyServer.dir/__/__/__/third_party/tracy/server/TracyColor.cpp.o
  [282/348] Building CXX object tracy/CMakeFiles/IREETracyCommon.dir/__/__/__/third_party/tracy/public/common/tracy_lz4.cpp.o
  [283/348] Linking CXX static library tracy/libIREETracyCommon.a
  [284/348] Building CXX object tracy/CMakeFiles/IREETracyServer.dir/__/__/__/third_party/tracy/server/TracyBadVersion.cpp.o
  [285/348] Building CXX object tracy/CMakeFiles/IREETracyServer.dir/__/__/__/third_party/tracy/server/TracyEventDebug.cpp.o
  FAILED: tracy/CMakeFiles/IREETracyServer.dir/__/__/__/third_party/tracy/server/TracyEventDebug.cpp.o
  /opt/rh/devtoolset-10/root/usr/bin/c++  -I/main_checkout/iree/build_tools/third_party/tracy/../../../third_party/tracy/imgui -I/usr/include/capstone -I/usr/include/capstone/capstone -O3 -DNDEBUG -fPIC -Wno-unused-result -std=gnu++17 -MD -MT tracy/CMakeFiles/IREETracyServer.dir/__/__/__/third_party/tracy/server/TracyEventDebug.cpp.o -MF tracy/CMakeFiles/IREETracyServer.dir/__/__/__/third_party/tracy/server/TracyEventDebug.cpp.o.d -o tracy/CMakeFiles/IREETracyServer.dir/__/__/__/third_party/tracy/server/TracyEventDebug.cpp.o -c /main_checkout/iree/third_party/tracy/server/TracyEventDebug.cpp
  /main_checkout/iree/third_party/tracy/server/TracyEventDebug.cpp:6:10: fatal error: ../common/TracyQueue.hpp: No such file or directory
      6 | #include "../common/TracyQueue.hpp"
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
  compilation terminated.
  [286/348] Building CXX object tracy/CMakeFiles/IREETracyServer.dir/__/__/__/third_party/tracy/server/TracyImGui.cpp.o
  [287/348] Linking CXX shared module runtime/bindings/python/iree/_runtime.cpython-37m-x86_64-linux-gnu.so
  ninja: build stopped: subcommand failed.
```

Comment (ScottTodd) [reply build_tools/third_party/tracy/CMakeLists.txt:118]: Ah, looks like we only set `-DIREE_BUILD_TRACY=ON` in the releases, and that is failing.

**Pull request merged**